backend/infrahub/git/models.py

```diff
@@ -92,7 +93,8 @@ class GitRepositoryMerge(BaseModel):
     source_branch: str = Field(..., description="The source branch")
     destination_branch: str = Field(..., description="The destination branch")
     destination_branch_id: str = Field(..., description="The ID of the destination branch")
-    default_branch: str = Field(..., description="The default branch in Git")
+    default_branch: str | None = Field(default=None, description="The default branch in Git")
+    repository_type: RepositoryTypes = Field(..., description="The type of the repository.")
```

Comment:
In other places we use `repository_kind: str = Field(..., description="The kind of the repository")`. Could this serve the same purpose?